### PR TITLE
Update member dropdown

### LINKED ISSUE #2556: Introduce new switchable member dropdown
## Scope

- [ ] Implement changes to member dropdown, allowing to switch between member and channel
- [ ] Add animation for above
- [ ] Add animation for switching channel in the new flow
- [ ] Display balance breakdown in a tooltip

---

## Context
Faucet gives spendable 10 tJoys while new membership gets 100 of locked funds (spending only on transaction fees allowed).
Also channel balance is not visible to the viewer 

## Scope
* [ ] Locked due to faucet constraints (other locks tbc) Details on what is possible to do with locked funds (Invitation lock, for all JOY that are received from faucet or when other member invites you). https://github.com/Joystream/atlas/issues/3090
* [ ] Display channel balance in viewers top nav
* [ ] Suggested to show both member and channel balance at the same time (not sum it up, but show next to each other)
* [ ] https://github.com/Joystream/atlas/issues/3085

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201958687417145/1202736674734968) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by drillprop**
Fix #2556 
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=3201%3A111990

### CODE REVIEW

**Comment by kdembler** [comment]
Can't test right now because of a failed build but when checking it out yesterday I wasn't able to initialize funds transfer, the link would always open dialog for withdrawing funds from channel, even if I was in the "member mode"

**Comment by drillprop** [comment]
> Can't test right now because of a failed build but when checking it out yesterday I wasn't able to initialize funds transfer, the link would always open dialog for withdrawing funds from channel, even if I was in the "member mode"

fixed

File: packages/atlas/src/components/_overlays/MemberDropdown/MemberDropdownNav.styles.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  height: 100%;
+  background-color: ${cVar('colorBackgroundAlpha')};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`
+
+export const StyledIconWrapper = styled(IconWrapper)`
+  position: absolute;
+  right: -8px;
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_overlays/MemberDropdown/MemberDropdownNav.styles.ts]: I think we should use `sizes` here

File: packages/atlas/src/components/_overlays/MemberDropdown/MemberDropdownNav.styles.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  height: ${({ height }) => (height ? (typeof height === 'number' ? `${height}px` : height) : 'unset')};
+  width: ${({ width }) => (width ? (typeof width === 'number' ? `${width}px` : width) : 'unset')};
+`
+
+export const UserBalance = styled.div`
+  display: grid;
+  grid-auto-flow: column;
+  grid-auto-columns: max-content;
+  align-items: center;
+  gap: 5px;
```
Comment (rafalpawlow) [inline packages/atlas/src/components/_overlays/MemberDropdown/MemberDropdownNav.styles.ts]: I know this is not your change, but it should be `4px`, let's also change it to `sizes`

**Review by rafalpawlow — CHANGES_REQUESTED** [review]
Works pretty well, few small issues

- [x] this icon seems to have wrong size
Zrzut ekranu 2022-09-1 o 12 23 26
Zrzut ekranu 2022-09-1 o 12 23 43

- [x] tooltip has wrong width
Zrzut ekranu 2022-09-1 o 12 39 29
Zrzut ekranu 2022-09-1 o 12 40 31

- [x] we agreed to add `interactive` flag to the tooltip I mentioned above
- [x] when I touch tooltip on mobile, it fires links and buttons beneath (not sure if we can do something about it)
- [ ] do you think we could add some offset to this tooltip like it's on the design? Currently, it's a bit of a meld with dropdown for me

Zrzut ekranu 2022-09-1 o 12 56 29

**Comment by drillprop** [comment]
> * [ ]  do you think we could add some offset to this tooltip like it's on the design? Currently, it's a bit of a meld with dropdown for me

What do you mean? The offset is set to 8px by default

**Comment by rafalpawlow** [comment]
@drillprop Now it is ok, but before the tooltip was exactly in the middle of the dropdown. Have you changed anything in this regard?

**Comment by rafalpawlow** [comment]
@drillprop One more question. Shouldn't we show channel variant as default in studio?

**Comment by drillprop** [comment]
> @drillprop Now it is ok, but before the tooltip was exactly in the middle of the dropdown. Have you changed anything in this regard?

We had this offset set previously as well, but yeah I did a small change.
The tooltip is appended to MemberDropdown container now, not the HTML element which keeps the account balance inside https://github.com/Joystream/atlas/pull/3187/commits/1df6e12b99c2e1fe7156dd0475d03d1539fb9efd
(check the `containerRef` prop)
> @drillprop One more question. Shouldn't we show channel variant as default in studio?

Yeah, I guess you're right. I'll change that.

**Review by toiletgranny — CHANGES_REQUESTED** [review]
@drillprop It felt soo good to take this component for a spin! The animations you implemented are buttery smooth! 🥰

1. **As is:** Account balance tooltip appears upon hovering over this entire area. This causes accidental tooltip triggers. 
    
**To be:** Ideally, the tooltip is triggered when hovering specifically over the numerical value and the JOY icon.

2. **As is:** In the total balance tooltip, JOY token amounts are displayed using the `--typography-desktop-t200` CSS variable. Also, the the contents of each row in the breakdown is not centered vertically.
**To be:** JOY token amounts should be displayed using the `--typography-desktop-t200-strong` CSS variable, and there should be `align-items: center` applied on all rows in the breakdown (make sure the "[lock-icon] Invitation lock" label remains centered vertically after applying this change).

4. **As is:** I can't seem to be able to scroll down the contents of the member dropdown.
    
**To be:** I should be able to do so.

5. **As is:** When in Studio, the Member dropdown component opens up in the Member mode by default.
**To be:** When in Studio, the Member dropdown component should open up in the Channel mode by default.

6. **As is:** Having just one channel created, I see the "Switch channel" list item in the Channel mode of the Member dropdown component. 
    
**To be:** As long as I have just one channel, I should see the "Add new channel" list item instead of the "Switch channel" [(documentation reference)](https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-components?node-id=3201%3A114961).

7. All transition animations seem to fire with a slight delay, and rather than animating from 0% to 100%, they go from ~25% to 100% only. This is best inspected if recorded on the screen and then played in slow motion. Happy to jump on a call to discuss details. 

8. **As is:** Hovering over the little icon indicators in the "Member dropdown switcher" component triggers tooltip visibility but does not lighten up the avatar and does nothing upon pressing.
   
   **To be:** Ideally, hover interactions should always reflect what can be clicked on. In this case, I'd opt for either not displaying the tooltip at all when hovering over the little icon in the corner, OR letting the user click on the entire area of the "Member dropdown switcher" instead of just the Avatar component itself, as it's implemented now. In the latter case, the Avatar should lighten up upon hovering over the little icon in the corner too.

9. **As is:** Having created a new channel from the "Switch channel" -> "Add new channel" menu (which I already pointed out is not correct if I have only one channel), opening the Member dropdown again I can see it still in the "Switch channel" mode (it should be in the "home" mode), and its height doesn't reflect the new list contents.
   
   **To be:** Opening the Member dropdown component should always default to the "home" screen of either the Channel mode or Member mode, depending on the experience. 

10. **As is:** Changing channel in the Viewer experience doesn't do anything. 
    **To be:** It should set the active channel to the one I clicked on. This will be important once we implement the YouTube Partner Program landing page.

**Comment by toiletgranny** [comment]
@drillprop Thanks a lot! I can see that points 2 and 4 are not 100% addressed yet, so let me clarify:
- For the 2nd point, I meant applying this bolder text style token to all JOY token values on the right-hand side. Oh, and since we're here, is there a way to disable the above tooltip now that we have a richer one below? It feels very odd to have two tooltips for the same thing. 😅 
CleanShot 2022-09-02 at 15 52 04
   - Can we also make sure the lock icon and the "Invitation lock" label remain vertically centered in their surrounding container? 🙏
- For the 4th point, we're halfway there, but what I meant is that the Member dropdown should _always_ default to either the Member or Channel mode, based on the experience. This means that if I open up the Member dropdown in the Viewer experience, change its mode to Channel, close it and then reopen — ideally, I see it in the Member mode again because of the fact I'm still in the Viewer experience. Hope it makes sense! :D

**Comment by drillprop** [comment]
> Can we also make sure the lock icon and the "Invitation lock" label remain vertically centered in their surrounding container?

It's already centered. I don't think I can center this lock more :P 

Besides that, everything should be fixed.

**Review by rafalpawlow — APPROVED** [review]

**Pull request merged**